### PR TITLE
refactor: vizBuilder function signature

### PR DESCRIPTION
**Pull request by darrenjaneczek**
**What is this feature?**

Refactors the `vizBuilder` signature to not require the `def` object.
Consumers of `vizBuilder` can apply the same unit and title from the `def`, as desired.

**Which issue(s) does this PR fix?**:

- Progress toward #78183 

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/trails/AutomaticMetricQueries/graph-builders/heatmap.ts [hunk 1]
```diff
@@ -1,18 +1,12 @@
...
 import { HeatmapColorMode } from 'app/plugins/panel/heatmap/types';
 
-import { AutoQueryDef } from '../types';
-
-export function heatmapGraphBuilder(def: AutoQueryDef) {
-  return PanelBuilders.heatmap()
-    .setTitle(def.title)
-    .setUnit(def.unit)
-    .setOption('calculate', false)
-    .setOption('color', {
```
Comment (darrenjaneczek) [inline public/app/features/trails/AutomaticMetricQueries/graph-builders/heatmap.ts:11]: Prettier really doesn't like this vertical pattern...

File: public/app/features/trails/AutomaticMetricQueries/AutoVizPanel.tsx [hunk 2]
```diff
@@ -51,7 +51,9 @@ export class AutoVizPanel extends SceneObjectBase<AutoVizPanelState> {
 
   private getVizPanelFor(def: AutoQueryDef) {
     return def
-      .vizBuilder(def)
+      .vizBuilder()
+      .setTitle(def.title)
+      .setUnit(def.unit)
```
Comment (torkelo) [inline public/app/features/trails/AutomaticMetricQueries/AutoVizPanel.tsx:56]: the title and unit is already part of the def object, so the vizBuilder function should be able to access them (if they close them in a closure)

File: public/app/features/trails/AutomaticMetricQueries/query-generators/common/queries.ts [hunk 3]
```diff
@@ -14,35 +14,29 @@ export function getGeneralBaseQuery(rate: boolean) {
...
 
-  const main = createMainQuery(baseQuery, agg, unit);
-
-  const breakdown = createBreakdownQuery(baseQuery, agg, unit);
-
-  return { preview: main, main: main, breakdown: breakdown, variants: [] };
-}
-
-function createMainQuery(baseQuery: string, agg: string, unit: string): AutoQueryDef {
-  return {
```
Comment (darrenjaneczek) [inline public/app/features/trails/AutomaticMetricQueries/query-generators/common/queries.ts:25]: Removed the separate functions to take advantage of local common struct.

File: public/app/features/trails/AutomaticMetricQueries/query-generators/bucket/index.ts [hunk 4]
```diff
@@ -23,28 +23,28 @@ function generator(metricParts: string[]) {
...
-    vizBuilder: simpleGraphBuilder,
+    vizBuilder: simpleGraphBuilder(common),
   };
 
   const breakdown: AutoQueryDef = {
     ...common,
     variant: 'p50',
     queries: [percentileQuery(50, [VAR_GROUP_BY_EXP])],
-    vizBuilder: simpleGraphBuilder,
+    vizBuilder: simpleGraphBuilder(common),
```
Comment (torkelo) [inline public/app/features/trails/AutomaticMetricQueries/query-generators/bucket/index.ts:33]:
```suggestion
    vizBuilder: () => simpleGraphBuilder(breakdown),
```

I was thinking more along this line

**Review by torkelo — APPROVED** [review]
I had in mind a different solution but not sure which is best :)

File: public/app/features/trails/AutomaticMetricQueries/query-generators/bucket/index.ts (hunk 4 above)
Comment (darrenjaneczek) [reply public/app/features/trails/AutomaticMetricQueries/query-generators/bucket/index.ts:33]: I couldn't find a way to work around:
> Block-scoped variable 'breakdown' used before its declaration.ts(2448)

...besides using `common` instead.

Comment (darrenjaneczek) [reply public/app/features/trails/AutomaticMetricQueries/query-generators/bucket/index.ts:33]: I took another look and went down this path in:
- #80170
I prefer how it makes the builder functions not require their own lambda.

**Pull request merged**